src/shared/SettingsStore.ts

```diff
@@ -67,20 +78,66 @@ export class SettingsStore<T extends object> {
                     });
                 }
 
-                if (typeof v === "object" && v !== null && !Array.isArray(v))
-                    return self.makeProxy(v, root, `${path}${path && "."}${key}`);
+                if (typeof v === "object" && v != null && !v[SYM_IS_PROXY]) {
```

Comment:
nit: this can never be undefined at that point so === is more accurate :P

```suggestion
                if (typeof v === "object" && v !== null && !v[SYM_IS_PROXY]) {
```